Off-Top-App/off-top-flutter
# OT-FrontEndNotificatons | Got firebase to send notifcations to the emulator/flutter application

**What I did:**

- I am able to trigger the notification from the firebase console to the emulator.

**How to test:**

- Log into the Firebase web console using the credentials found in this [documentation](https://docs.google.com/document/d/1kqS7te_Rzmt-06OILNQuTDpwEqzwJCJ0VYcxnONbMy4/edit)

- Once logged in, click on the `off-top-firebase` box.

- In the navigation tab on the left, navigate to the `Grow` drop down menu.

- In there, click on `cloud messaging`.

- Launch the Off-Top application.

- Log into the app.

- In the flutter terminal there will be a `FirebaseMessaging token` you need to copy.

![image](https://user-images.githubusercontent.com/56651529/87118447-44665f80-c230-11ea-96bf-84c231f32006.png)

- In your Firebase window click on `new notification`.

- Enter a `Notification tile `and `Notification Text`.

- Click next and choose the targeting app to be `Off Top Android`.

- Click next and chose the scheduling to be `Now`.

- Click next until you get to step 5, enter the following information to match the photo below

![image](https://user-images.githubusercontent.com/56651529/87118841-467cee00-c231-11ea-9310-0af8ca997715.png)

- Go back to step 1 shown below 

![image](https://user-images.githubusercontent.com/56651529/87118932-7f1cc780-c231-11ea-9add-ba007b35b308.png)

- Click on `send test message` and paste the token from the terminal here and add it by pressing the plus icon.

![image](https://user-images.githubusercontent.com/56651529/87119079-d1f67f00-c231-11ea-9b9d-8bb2ecab8296.png)

- Click `Test` and when the app is either in the background or foreground the notification will be sent but when the app is in the foreground you should also see the message in the terminal with the data shown in the picture.

**Foreground**

![image](https://user-images.githubusercontent.com/56651529/87119456-e1c29300-c232-11ea-8953-15e1a4a35467.png)

**Background**

![image](https://user-images.githubusercontent.com/56651529/87119513-1d5d5d00-c233-11ea-9124-1e1a9da91c4c.png)

- If you want to test another message notification you can create a new one by clicking `new notification` or you can click on the 3 dots of your last notification and click on duplicate to send another test message by following the same steps above.
